src/lib.rs

```diff
@@ -21,6 +23,24 @@ use std::fs::OpenOptions;
 use std::io::Write;
 use std::path::PathBuf;
 
+#[derive(Debug, StructOpt)]
```

Comment:
I don't see why we would put an option parser into the library portion. The main function is where option parsing belongs.